dhis-2/dhis-api/src/main/java/org/hisp/dhis/expression/ExpressionService.java

```diff
@@ -275,59 +273,7 @@ void getExpressionDimensionalItemMaps( String expression, ParseType parseType, D
-
-    /**
-     * Generates the calculated numeric value for an expression.
-     *
-     * @param expression the expression string.
-     * @param parseType the type of expression to parse.
-     * @param itemMap map of dimensional item objects to value in expression.
-     * @param valueMap the dimensional item object values to use for
-     *        calculation.
-     * @param constantMap map of constants to use for calculation.
-     * @param orgUnitCountMap the map of organisation unit group member counts.
-     * @param orgUnitGroupMap the map of organisation unit groups.
-     * @param days the number of days to use in the calculation.
-     * @param missingValueStrategy the strategy to use when data values are
-     *        missing when calculating the expression.
-     * @param orgUnit the current organisation unit.
-     * @return the calculated value as a double.
-     */
-    Double getExpressionValue( String expression, ParseType parseType,
-        Map<DimensionalItemId, DimensionalItemObject> itemMap, Map<DimensionalItemObject, Object> valueMap,
-        Map<String, Constant> constantMap, Map<String, Integer> orgUnitCountMap,
-        Map<String, OrganisationUnitGroup> orgUnitGroupMap, Integer days,
-        MissingValueStrategy missingValueStrategy, OrganisationUnit orgUnit );
-
-    /**
-     * Generates the calculated value for an expression.
-     *
-     * @param expression the expression string.
-     * @param parseType the type of expression to parse.
-     * @param itemMap map of dimensional item objects to value in expression.
-     * @param valueMap the dimensional item object values to use for
-     *        calculation.
-     * @param constantMap map of constants to use for calculation.
-     * @param orgUnitCountMap the map of organisation unit group member counts.
-     * @param orgUnitGroupMap the map of organisation unit groups.
-     * @param days the number of days to use in the calculation.
-     * @param missingValueStrategy the strategy to use when data values are
-     *        missing when calculating the expression.
-     * @param orgUnit the current organisation unit.
-     * @param samplePeriods periods for samples to aggregate.
-     * @param periodValueMap values for aggregate functions by period.
-     * @param dataType the data type the expression should return.
-     * @return the calculated value.
-     */
-    Object getExpressionValue( String expression, ParseType parseType,
-        Map<DimensionalItemId, DimensionalItemObject> itemMap, Map<DimensionalItemObject, Object> valueMap,
-        Map<String, Constant> constantMap, Map<String, Integer> orgUnitCountMap,
-        Map<String, OrganisationUnitGroup> orgUnitGroupMap, Integer days,
-        MissingValueStrategy missingValueStrategy, OrganisationUnit orgUnit, List<Period> samplePeriods,
-        MapMap<Period, DimensionalItemObject, Object> periodValueMap, DataType dataType );
+    Object getExpressionValue( ExpressionParams exParams );
```

Comment:
Javadoc now misses the `@return` which given we return an `Object` is important to describe even beyond the old text ;)

Reply:
Ooops, done. I was too excited because I'd removed all the other params.